src/Adapter/Driver/AbstractConnection.php

```diff
@@ -41,39 +32,24 @@ public function disconnect()
         return $this;
     }
 
-    /**
-     * Get connection parameters
-     *
-     * @return array
-     */
-    public function getConnectionParameters()
+    /** Get connection parameters */
+    public function getConnectionParameters(): array
     {
         return $this->connectionParameters;
     }
 
-    /**
-     * Get driver name
-     *
-     * @return null|string
-     */
-    public function getDriverName()
+    /** Get driver name */
+    public function getDriverName(): ?string
```

Comment:
What about setting the return type of this and others to be a union type, e.g., `string|null` to avoid the deprecation warning in 8.4?

Reply:
I'll look into it. Didn't realize it was being deprecated. Can you please link me to the information about it. I checked the 8.4 deprecations and unless I just missed it cause I'm on mobile I didn't see it in the list.

Thanks

Reply:
Ok, so I am pretty sure you are thinking of the implicitly nullable parameter deprecation, which I am aware of which is what confused me.

1. The deprecation in 8.4 does not affect return types
2. It also does not require a union.

Please see:
https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated#change

Reply:
I thought that it did affect return types. My bad.